datafusion/execution/src/memory_pool/mod.rs

```diff
@@ -175,64 +174,6 @@ impl Drop for MemoryReservation {
     }
 }
 
-pub trait TryGrow: Send + Sync + std::fmt::Debug {
```

Comment:
As these are relatively new I figured it was easiest to just remove them, rather than go through a deprecation process